sonic_sfp/sfputilbase.py

```diff
@@ -198,7 +198,7 @@ def _sfp_eeprom_present(self, sysfs_sfp_i2c_client_eeprompath, offset):
             return False
         else:
             try:
-                sysfsfile = open(sysfs_sfp_i2c_client_eeprompath, "rb")
+                sysfsfile = open(sysfs_sfp_i2c_client_eeprompath, mode="rb", buffering=0)
```

Comment:
>open(sysfs_sfp_i2c_client_eeprompath, mode="rb", buffering=0) [](start = 28, length = 61)

Encapsulate a function like: self.open_sysfsfile()

Reply:
I looked at the code structure. It makes more sense to create a help function to return opened file and validate an offset at the same time. Because every place this function is called, there is another file opening following.